src/Serializer/XlsxEncoder.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace AppBundle\Serializer;
+
+use PhpOffice\PhpSpreadsheet\Spreadsheet;
+use PhpOffice\PhpSpreadsheet\Writer\Xlsx;
+use Symfony\Component\Serializer\Encoder\EncoderInterface;
+
+class XlsxEncoder implements EncoderInterface
+{
+    public const FORMAT = 'xlsx';
+
+    public function encode($data, $format, array $context = [])
+    {
+        if (!is_iterable($data)) {
+            throw new \InvalidArgumentException('This method requires a collection of entities');
```

Comment:
normally it cannot be entities here, as we are in encoding process 
`This method requires an array of data`?